mage_integrations/mage_integrations/destinations/snowflake/__init__.py

```diff
@@ -334,6 +298,63 @@ def calculate_records_inserted_and_updated(
 
         return records_inserted, records_updated
 
+    def process_queries(
+        self,
+        query_strings: List[str],
+        record_data: List[Dict],
+        stream: str,
+        tags: Dict = None,
+        **kwargs,
+    ) -> List[List[Tuple]]:
+        if not self.use_batch_load:
+            self.logger.info('Using default SQL insertion load for Snowflake...')
+            return super(Snowflake, self).process_queries(
+                query_strings,
+                record_data,
+                stream,
+                tags,
+                **kwargs,
+            )
+        else:
+            results = []
+            self.logger.info('Using batch load for Snowflake...')
+            tries = tags.get('tries', 0)
+
+            try:
+                df = pd.DataFrame([d['record'] for d in record_data])
+                success, num_chunks, num_rows, output = write_pandas(
+                    self.build_connection().connection,
+                    df,
+                    tags.get('table_name'),
```

Comment:
I think we'll need to insert the data into a temp table first and then merge into the main table since we need to handle the merge conflict. https://github.com/mage-ai/mage-ai/blob/master/mage_integrations/mage_integrations/destinations/snowflake/__init__.py#L231-L235

Reply:
> Let's get table_name, database_name, schema_name from self.config instead of tags.

Made the suggested change.

Reply:
> I think we'll need to insert the data into a temp table first and then merge into the main table since we need to handle the merge conflict. https://github.com/mage-ai/mage-ai/blob/master/mage_integrations/mage_integrations/destinations/snowflake/__init__.py#L231-L235

@wangxiaoyou1993 I've added the necessary handling for merging related process. Please review the latest updates.

Is merging required all the time or only when the following conditions are met:
```
if unique_constraints and unique_conflict_method:
``` 
This condition is used to check whether to use merging in the Base class.

Reply:
Yes, that's the condition when the merge is required.